apps/sms/test/marionette/composer_test.js

```diff
@@ -257,38 +258,73 @@ marionette('Messages Composer', function() {
   });
 
 
-  suite('Invalid recipients', function() {
+  suite('Recipients list', function() {
     var newMessage;
+    var storage;
+    var contact = {
+      name: ['Existing Contact'],
+      givenName: ['Existing'],
+      familyName: ['Contact'],
+      tel: [{
+        value: '5551234567',
+        type: 'Mobile'
+      }]
+    };
 
     setup(function() {
-      messagesApp.launch();
+      var MOCKS = [
+        '/mocks/mock_test_storages.js',
+        '/mocks/mock_navigator_moz_contacts.js'
+      ];
+      storage = Storage.create(client);
+
+      MOCKS.forEach(function(mock) {
+        client.contentScript.inject(__dirname + mock);
+      });
 
+      messagesApp.launch();
+      storage.setMessagesStorage();
+      storage.setContactsStorage([contact]);
       var inbox = new InboxView(client);
       newMessage = inbox.createNewMessage();
     });
 
-    suite('the recipients list', function() {
-      test('should display that a non existing contact is invalid', function() {
-        newMessage.addNewRecipient('non_exisiting_contact');
-        assert.isTrue(newMessage.containsInvalidRecipients());
+    teardown(function() {
+      storage.setContactsStorage();
+    });
+
+
+    suite('The recipients', function() {
```

Comment:
My bad, wrong merge during the cherry pick. Here's what we get now:
>     Recipients
      ✓ should match an existing contact
      Invalid recipients
        Recipients list
          ✓ should display that a non existing contact is invalid (1859ms)
          ✓ should allow to correct an invalid contact (4490ms)
        Content composer
          ✓ should not enable send button if the contact is invalid (2553ms)